setup.py

```diff
@@ -143,14 +143,15 @@ def _collect_requirements(requires):
     "compress-pickle[lz4]~=2.1.0",
     "coverage[toml]",
     "dask[complete]~=2021.10",  # pinned by Snyk to avoid a vulnerability
-    "delta-spark~=1.2.1",  # 1.2.0 has a bug that breaks some of our tests: https://github.com/delta-io/delta/issues/1070
+    "delta-spark>=1.2.1; python_version >= '3.11'",  # 1.2.0 has a bug that breaks some of our tests: https://github.com/delta-io/delta/issues/1070
+    "delta-spark~=1.2.1; python_version < '3.11'",
```

Comment:
Doesn't `delta-spark>=1.2.1` mean the same as `delta-spark~=1.2.1`? https://peps.python.org/pep-0440/#compatible-release

Reply:
From what I understand `delta-spark>=1.2.1` will allow `python 3.11` to use any future version of delta-spark even outside the 1.2.x series such as 1.3.x etc. While `delta-spark~=1.2.1` will limit all other python versions to only anything in the 1.2.x series starting from `1.2.1`.

Reply:
The `~` syntax still confuses me, but what you're saying sounds right.